### PR TITLE
add porting in a nutshell car analogy

### PR DESCRIPTION
**Pull request by Danfro**
There have been questions in Welcome Group about what porting is in a nutshell. I wrote an explanation that was considered worth storing somewhere. So here it is. ;-) If you think of a better place for this - just say a word. If you think it needs to be added somewhere else too let me know.

That's what it would look like:

screenshot

### CODE REVIEW

File: porting/first-steps.rst
Comment (doniks) [inline porting/first-steps.rst:22]: may I suggest a few small adjustments:

You do have -> You have
a electrical -> an electrical
driven engine -> engine (two times)
you get told -> you are told

Comment (Danfro) [reply porting/first-steps.rst]: Thanks for the suggestions. 1, 2 and 4 are obvious. For 3 I changed the wording to " for your new engine " to avoid the double occurrence.

**Comment by Danfro** [comment]
Ok, thats what we have got now:

grafik

**Comment by Danfro** [comment]
Is "electric driven engine" a phrase that would actually be used? Online translators suggest just "electric engine" or "electric motor". Would one of those be better?

**Comment by Danfro** [comment]
And thanks for offering a place at the "front page". :smirk:

**Comment by doniks** [comment]
> Is "electric driven engine" a phrase that would actually be used? Online translators suggest just "electric engine" or "electric motor". Would one of those be better?

My sentiment exactly! I am not a native english speaker, but "electric engine" sounds more natural and logical to me. I suggest to change to "electric engine"

**Comment by Danfro** [comment]
Great, thanks for the feedback. I get that changed.

And I am not sure if the named car makes are worldwide known. Ford should be fine. Maybe we use Toyota or Ferrari or Mercedes as the other one?

Am Wed, 5 Aug 2020 22:31:59 -0700, doniks schrieb:
>> Is "electric driven engine" a phrase that would actually be 
>> used? Online translators suggest just "electric engine" or 
>> "electric motor". Would one of those be better?
>
> My sentiment exactly! I am not a native english speaker, but 
> "electric engine" sounds more natural and logical to me. I 
> suggest to change to "electric engine"
>

--

**Comment by doniks** [comment]
I do not have a strong opinion on the car brands :)

FWIW, I do not care about cars at all, but I do recognize all brands you mentioned ;) up to you!

**Comment by Danfro** [comment]
Cars are tools. ☺ The only concern I would have, is to find car makes that are widely known worldwide. Although when translating the text, car brands could be replaced if needed.

Am Thu, 6 Aug 2020 11:18:45 -0700, doniks schrieb:
> I do not have a strong opinion on the car brands :)
>
> FWIW, I do not care about cars at all, but I do recognize all 
> brands you mentioned ;) up to you!
>

--

**Comment by mariogrip** [comment]
As a suggestion maybe call Lineage OS a *similar petrol engine* as generally easier to port android to android.

**Comment by Danfro** [comment]
Almost forgot about your comment. 😳

Although I understand your point, I don't think using a *similar petrol engine* would make the point clear enough.

But maybe using diesel for Lineage and electic engine for UT is better. Diesel and petrol are more similar as both burn some sort of fuel. An electrical engine is then a very different thing.

Am Sat, 8 Aug 2020 14:44:37 -0700, Marius Gripsgard schrieb:
> As a suggestion maybe call Lineage OS a *similar petrol engine* 
> as generally easier to port android to android. 
>

--

**Comment by Danfro** [comment]
I should not have written that last comment on my phone. Lineage is already called diesel and UT is electric engine. :sweat_smile: 

@doniks Could you give this another review please? It seems to be stuck there. Or do you need something else changed? Maybe after the time since creating this MR I am missing something...
How is your opinion regarding Marius' suggestion?

**Review by doniks — APPROVED** [review]

**Comment by doniks** [comment]
I was basically only waiting on you to get *your* final approval that you are happy with the car brand choices :D I had the impression you were hesitant and questioning your draft.

I'm not so much into splitting hairs about the fine lines here to be honest. I think the analogy is clear and there is little value in over engineering. That said. If someone (you, Marius, anybody) feels strongly about the types of fuel or the brands used in this version ... PR's always have been and still are welcome :)

**Pull request merged**
